design/collision.layers.v2.md

```diff
@@ -0,0 +1,60 @@
+ :--|:-:|:-:|:-:|:-:|--:
+ Physical|Rigid body, Player Collision, Blocks Cursor|no*|Default|Default, NavMesh, CharacterController, Interactables|no
+ PhysicalTeleportable|Same as Physical, but is also valid teleport destination|no*|NavMesh|Default, NavMesh, CharacterController, Interactables|no
+ CharacterBlocking|blocks charactercontroller, but not rigid bodies or cursor|yes|IgnoreRaycast|CharacterController|no
+ UI|ButtonBehavior,ToolBehavior, 2D UI|yes|MREUI(new)|None|yes
+ PersonalTrigger|Trigger Volumes for charactercontrollers and exclusive (not group masked) rigid bodies|yes|MREPersonalTrigger(new)|Default,NavMesh,CharacterController|yes
+ GlobalTrigger|Trigger Volumes for global (not group masked/exclusive) rigid bodies|no|MREGlobalTrigger(new)|Default,NavMesh,CharacterController|yes
+ 
+*Note about physical/physicalteleportable: Rigid bodies (but not static or kinematic objects with colliders) are allowed to be exclusive actors (but not groupmasked), because by setting mass super low, the global actors always push exclusive actors. Therefore the exclusive rigid bodies don't mess with the global consistency of the physics simulation. So you can stack exclusive actors on global actors, but not vice versa.
+
+Note that PersonalTrigger and PhysicalTrigger code will have to manually filter out actors that is in the right unity layer, but don't match the requirements (personaltrigger=must be client's own character controller, or exclusive actors, but exclude other characters and groupmasked/global actors, physicaltrigger must exclude groupmasked/exclusive actors).
+
+## Notes
+
+Altspace details: While I know Altspace is layer constrained, I think I have come up with 3 layers.
+
+We would remove the “IsTrigger” flag from the MRE api surface – it’s implicit from MRE collision layer now.
+
+Any category of messages (collisionOn/collisionOff/behaviors) that are allowed to be non-global should be sent from each client, not by authoritative peer.
+
+Required Error checking: 
+1. We should error on physical/physicalandteleportable/physicaltrigger collision layers assigned to exclusive/grouped actors that are in the wrong groups. We should also error if user changes collision layer after actor creation on physical/physicalandteleportable/physicaltrigger collision layers assigned to exclusive/grouped actors.
+2. We should warn when button (possibly certain tool) behavior is using layer other than UI
+
+## Implementation order
+1. add new layers to sdk and runtime, and configure testbeds and altspace implementations, including the collision masks
+2. remove istrigger flag from MRE API and protocol, and drive it from Unity Runtime
+3. enforce errors on actor creation
+4. enforce errors on collision layer change after actor creation
+5. check for warning on button behavior assignment
+6. Make triggers filter out the events that don't match the requirements
+
```

Comment:
1) should add this layer support to the prediction code, as far as I know, the jitter buffer should not know about the collision layers. 
If not added would not be a show stopper but could eliminate unnecessary artifacts.